### ISSUE TITLE
MD032: Markdown tables as list continuation content breaks `rumdl` list detection

### ISSUE THREAD

**Issue by eread** [rvben/rumdl#533]
### rumdl version

0.1.51

### Installation method

mise

### Bug description

Rule MD032 (https://github.com/rvben/rumdl/blob/79a29f34ace341acdf6109777c3c22fcef0cfb6e/docs/md032.md) ensures readability by requiring blank lines around lists.

However, if a list contains an indented table, `rumdl` looses track of where the list should start and end.

### Steps to reproduce

1. Create a Markdown file with this content:
   ```markdown
   ---
   title: Level 1 heading
   ---
   
   - List item 1.
     - Nested list item
   
     | Table col 1 | Table col 2 |
     |:------------|:------------|
     | Table row 1 | Table row 1 |
   
   - List item 2.
   ```
1. Run `rumdl check` on Markdown file.

### Expected behavior

Given the scenario above, I'd expect no violations.

### Actual behavior

Given the scenario above, `rumdl` returns the following:

```shell
test.md:8:1: [MD032] List should be followed by blank line [*]
```

### Configuration

None, use default settings.

### Operating system

macOS

**Comment by rvben** [comment]
Fixed in v0.1.52. Indented tables within list items are now correctly treated as list continuation content, so MD032 no longer produces false positives in this case. This was also verified against the kubernetes/website repo which had the same pattern.

Thanks for reporting this @eread!
